app/codes/transactionmanager.py

```diff
@@ -438,6 +438,9 @@ def econvalidator(self):
             if not wallet1valid or not wallet2valid:
                 print("One of the wallets is invalid")
                 self.validity = 0
+            elif wallet1 == wallet2:
```

Comment:
the comparison should be of pids ..
since one pid can have multiple wallet, this check is not sufficient.

Reply:
yes. haven't thought of that. was only checking single wallet score.